pkg/service/bigquery/bigquery_client.go

```diff
@@ -260,24 +231,9 @@ func (d *defaultBigQueryClient) GetTablePartitions(ctxIn context.Context, projec
 		return nil, fmt.Errorf("GetTablePartitions failed for `%s.%s.%s`, because partition other then DAY is not supported", project, dataset, table)
 	}
 
-	timePartitioningField := "_PARTITIONTIME"
-	targetFieldInTablePartition := "T_TIMESTAMP"
-	if metadata.TimePartitioning.Field != "" {
-		timePartitioningField = metadata.TimePartitioning.Field
-		for _, schema := range metadata.Schema.Relax() {
-			if schema.Name == metadata.TimePartitioning.Field {
-				targetFieldInTablePartition = fmt.Sprintf("T_%s", schema.Type)
-				break
-			}
-		}
-	}
-
 	var partitions []*Table
-	q := fmt.Sprintf("SELECT COUNT(*) AS total, %s AS %s FROM `%s.%s.%s` WHERE %s IS NOT NULL GROUP BY %s",
-		timePartitioningField, targetFieldInTablePartition,
+	q := fmt.Sprintf("SELECT total_rows AS total, partition_id FROM `%s.%s.INFORMATION_SCHEMA.PARTITIONS` WHERE TABLE_NAME = '%s'",
```

Comment:
have you tested how does looks partition_id for custom partition column filed with different types (TIMESTAMP, DATE, DATETIME, TIME) ?

it can be a possible breaking change

Reply:
I created 3 table in BigQuery with Timestamp, Date and Datetime for Partition column. It works for all three. Time does not make sense since we only support day partitioning (see code above). 

You can find the tables here:
https://console.cloud.google.com/bigquery?project=oghub-acme-b-d

Penelope did backups successfully here:
https://console.cloud.google.com/storage/browser/bkp_bq_bc13acf7-795a-4c34-a282-bd729c776670/dataset/deleteme_penelope_test/table;tab=objects?pageState=(%22StorageObjectListTable%22:(%22f%22:%22%255B%255D%22))&project=oghub-acme-backup&prefix=&forceOnObjectsSortingFiltering=false